sdk/python/lib/pulumi/runtime/resource.py

```diff
@@ -154,7 +154,7 @@ async def prepare_resource(
     # For remote resources, merge any provider opts into a single dict, and then create a new dict with all of the
     # resolved provider refs.
     provider_refs: Dict[str, Optional[str]] = {}
-    if remote and opts is not None:
+    if (remote or not custom) and opts is not None:
```

Comment:
This is just "not custom" now isn't it? Resources are either LocalComponent, RemoteComponent, or Custom (remote=true, custom=true should be inexpressible). Also is a similar change not needed in the dotnet code?

Reply:
I couldn't find the equivalent logic. It is just a branch on `custom`: https://github.com/pulumi/pulumi/blob/85cfd79d32647019c5edb6ff45130aff9c66d10a/sdk/dotnet/Pulumi/Resources/Resource.cs#L221-L259